library/src/main/java/com/novoda/downloadmanager/PathBasedFilePersistence.java

```diff
@@ -55,13 +56,15 @@ public FilePersistenceResult create(FilePath absoluteFilePath, FileSize fileSize
     }
 
     private boolean ensureParentDirectoriesExistFor(File outputFile) {
-        boolean parentExists = outputFile.getParentFile().exists();
-        if (parentExists) {
-            return true;
-        }
+        synchronized (LOCK) {
```

Comment:
there shouldn't be much of a performance hit for the sequential file downloads 🤞